dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ProcessRestRepository.java

```diff
@@ -135,6 +150,104 @@ protected void delete(Context context, Integer integer)
         }
     }
 
+    /**
+     * Search method that will take Parameters and return a list of {@link ProcessRest} objects
+     * based on the {@link Process} objects that were in the databank that adhere to these params
+     * @param ePersonUuid   The UUID for the EPerson that started the Process
+     * @param scriptName    The name of the Script for which the Process belongs to
+     * @param processStatusString   The status of the Process
+     * @param pageable      The pageable
+     * @return              A page of {@link ProcessRest} objects adhering to the params
+     * @throws SQLException If something goes wrong
+     */
+    @SearchRestMethod(name = "byProperty")
+    @PreAuthorize("hasAuthority('ADMIN')")
+    public Page<ProcessRest> findProcessesByProperty(@Parameter(value = "userId") UUID ePersonUuid,
+                                                     @Parameter(value = "scriptName") String scriptName,
+                                                     @Parameter(value = "processStatus") String processStatusString,
+                                                     Pageable pageable)
+        throws SQLException {
+        if (StringUtils.isBlank(scriptName) && ePersonUuid == null && StringUtils.isBlank(processStatusString)) {
+            throw new DSpaceBadRequestException("Either a name, user UUID or ProcessStatus should be provided");
+        }
+
+        Context context = obtainContext();
+        EPerson ePerson = epersonService.find(context, ePersonUuid);
+        if (ePersonUuid != null && ePerson == null) {
+            throw new DSpaceBadRequestException("No EPerson with the given UUID is found");
+        }
+
+        ProcessStatus processStatus = StringUtils.isBlank(processStatusString) ? null :
+            ProcessStatus.valueOf(processStatusString);
```

Comment:
what will happen if it's not a valid status, is there an IT for this use case?

Reply:
If an invalid status is given, the valueOf call will throw an exception which will in turn be parsed to a BadRequest status code given by the REST api. I've added a test to verify this here: https://github.com/DSpace/DSpace/pull/2931/commits/13b59f9522c752f66455f2b83d6f3f67971c81fb#diff-5c0eb5afe2a696f9d1a29faffc28ccc2R586